File: backend/src/main/java/com/awais/hr/module/leave/service/LeaveService.java
package com.awais.hr.module.leave.service;

import com.awais.hr.module.leave.dto.LeaveRequestDTO;
import com.awais.hr.module.leave.dto.LeaveStatusUpdateDTO;
import java.util.List;
import java.util.Map;

public interface LeaveService {
    List<Map<String, Object>> getPolicies();
    List<Map<String, Object>> getRequests(String email);
    void submitRequest(String email, LeaveRequestDTO dto);
    void updateRequestStatus(String approverEmail, String id, LeaveStatusUpdateDTO dto);
    void deleteRequest(String id);
}

File: frontend/src/services/leaveService.js
import apiClient from './api';

export const getPolicies = () => {
  return apiClient.get('/leaves/policies');
};

export const getRequests = () => {
  return apiClient.get('/leaves/requests');
};

export const submitRequest = (dto) => {
  return apiClient.post('/leaves/requests', dto);
};

export const updateRequestStatus = (id, dto) => {
  return apiClient.put(`/leaves/requests/${id}/status`, dto);
};

export const deleteRequest = (id) => {
  return apiClient.delete(`/leaves/requests/${id}`);
};
